### PR TITLE
Feature/statistics

### PR DESCRIPTION
**Pull request by Permas-Teo**
implement updates to charts and daily challenge in Statistics tab after each pom command.

TODO: 
update after Done command
update at init()

### CODE REVIEW

File: src/main/java/seedu/address/ui/StatisticsDisplay.java [hunk 1]
```diff
@@ -12,69 +14,129 @@
...
+        }
 
-        barChart.getData().add(dataSeries1);
-    }
+        switch (expBarPerc) {
+            case 0:
+                this.progressBarDailyFilepathString = "/images/pet/ProgressBar0%.png";
+                break;
+            case 1:
+                this.progressBarDailyFilepathString = "/images/pet/ProgressBar10%.png";
```
Comment (BransonNg) [inline src/main/java/seedu/address/ui/StatisticsDisplay.java:84]: maybe a String.format()?
String.format("/images/pet/ProgressBar%d0%.png", expBarPerc); 
Except for the 0 one, unless you rename the filename

File: src/main/java/seedu/address/model/dayData/CustomQueue.java [hunk 2]
```diff
@@ -243,30 +243,34 @@ public int hashCode() {
     }
 
     /** Returns true if {@code dayDatas} table constraints are enforced */
-    private boolean tableConstraintsAreEnforced(List<DayData> dayDatas) {
-        int count = 0;
+    public static boolean tableConstraintsAreEnforced(List<DayData> dayDatas) {
```
Comment (BransonNg) [inline src/main/java/seedu/address/model/dayData/CustomQueue.java:246]: Mmm, given that statistics is independent of user input, i feel like if the tableConstraints aren't meant, we shouldn't throw an error? We should just do an internal fix like reset data or recreate table based on remaining valid data

File: src/main/java/seedu/address/model/dayData/CustomQueue.java [hunk 3]
```diff
@@ -243,30 +243,34 @@ public int hashCode() {
...
+        boolean res = true;
```
Comment (BransonNg) [inline src/main/java/seedu/address/model/dayData/CustomQueue.java:247]: also, could replace all the res = false with exceptions? then you can state explicitly what the error was

**Review by BransonNg — APPROVED** [review]
LGTM!
Only the String.format in StatisticsDisplay and the Error handling for constraints in ModelManagerr

File: src/main/java/seedu/address/model/dayData/CustomQueue.java (hunk 3 above)
Comment (Permas-Teo) [reply src/main/java/seedu/address/model/dayData/CustomQueue.java:247]: I followed the format for TaskList, apparently this method is used to determine whether an exception should be thrown or not. I will check and get back to you again!

**Pull request merged**